stackslib/src/chainstate/stacks/boot/pox_4_tests.rs

```diff
@@ -1629,58 +1630,105 @@ fn delegate_stack_stx_signer_key() {
+    .expect_tuple();
+
+    let stacking_state = get_stacking_state_pox_4(
+        &mut peer,
+        &latest_block,
+        &key_to_stacks_addr(stacker_key).to_account_principal(),
+    )
+    .expect("No stacking state, delegate-stack-stx failed")
+    .expect_tuple();
+
+    let state_signer_key = stacking_state.get("signer-key").unwrap();
+
+    assert_eq!(
+        state_signer_key.to_string(),
+        format!("0x{}", signer_public_key.to_hex())
+    );
+}
+
+#[test]
+fn delegate_stack_stx_extend_signer_key() {
+    let lock_period: u128 = 2;
+    let (burnchain, mut peer, keys, latest_block, block_height, mut coinbase_nonce) =
+        prepare_pox4_test(function_name!(), None);
+
+    let mut stacker_nonce = 0;
+    let stacker_key = &keys[0];
+    let delegate_nonce = 0;
+    let delegate_key = &keys[1];
+    let delegate_principal = PrincipalData::from(key_to_stacks_addr(delegate_key));
+    let signer_private_key = &keys[2];
+    let signer_public_key = StacksPublicKey::from_private(signer_private_key);
+    let new_signer_private_key = &keys[3];
+    let new_signer_public_key = StacksPublicKey::from_private(new_signer_private_key);
+    let pox_addr = PoxAddress::from_legacy(
+        AddressHashMode::SerializeP2PKH,
         key_to_stacks_addr(delegate_key).bytes,
     );
-    let signer_key_val = Value::buff_from(vec![
-        0x03, 0xa0, 0xf9, 0x81, 0x8e, 0xa8, 0xc1, 0x4a, 0x82, 0x7b, 0xb1, 0x44, 0xae, 0xc9, 0xcf,
-        0xba, 0xeb, 0xa2, 0x25, 0xaf, 0x22, 0xbe, 0x18, 0xed, 0x78, 0xa2, 0xf2, 0x98, 0x10, 0x6f,
-        0x4e, 0x28, 0x1b,
-    ])
-    .unwrap();
 
-    let txs = vec![
-        make_pox_4_contract_call(
-            stacker_key,
-            stacker_nonce,
-            "delegate-stx",
-            vec![
-                Value::UInt(100),
-                delegate_principal.clone().into(),
-                Value::none(),
-                Value::Optional(OptionalData {
-                    data: Some(Box::new(pox_addr.clone())),
-                }),
-            ],
-        ),
-        make_pox_4_contract_call(
-            delegate_key,
-            delegate_nonce,
-            "delegate-stack-stx",
-            vec![
-                PrincipalData::from(key_to_stacks_addr(stacker_key)).into(),
-                Value::UInt(100),
-                pox_addr,
-                Value::UInt(block_height as u128),
-                Value::UInt(lock_period),
-                signer_key_val.clone(),
-            ],
-        ),
-    ];
-    // (define-public (delegate-stack-stx (stacker principal)
-    //                                (amount-ustx uint)
-    //                                (pox-addr { version: (buff 1), hashbytes: (buff 32) })
-    //                                (start-burn-ht uint)
-    //                                (lock-period uint)
-    //                                (signer-key (buff 33)))
+    let delegate_stx: StacksTransaction = make_pox_4_delegate_stx(
```

Comment:
I'd love to see human-readable descriptions of the tests in pox 4. It is already difficult enough to understand.